frontend/src/metabase/containers/AddToDashSelectDashModal/AddToDashSelectDashModal.unit.spec.tsx

```diff
@@ -141,59 +171,490 @@ describe("AddToDashSelectDashModal", () => {
+            const dashboardInPersonalCollection = createMockDashboard({
+              id: 5,
+              name: "Dashboard in personal collection",
+              collection_id: PERSONAL_COLLECTION.id as number,
+              model: "dashboard",
+            });
+
+            await setup({
+              dashboard: dashboardInPersonalCollection,
+            });
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: PERSONAL_COLLECTION.name,
+              }),
+            );
+
+            expect(
+              await screen.findByRole("heading", {
+                name: dashboardInPersonalCollection.name,
+              }),
+            ).toBeInTheDocument();
+          });
+
+          it("should render dashboards when opening personal subcollections", async () => {
+            const dashboardInPersonalSubcollection = createMockDashboard({
+              id: 5,
+              name: "Dashboard in personal subcollection",
+              collection_id: PERSONAL_SUBCOLLECTION.id as number,
+              model: "dashboard",
+            });
+
+            await setup({
+              dashboard: dashboardInPersonalSubcollection,
+            });
+
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: PERSONAL_COLLECTION.name,
+              }),
+            );
+            userEvent.click(
+              screen.getByRole("heading", {
+                name: PERSONAL_SUBCOLLECTION.name,
+              }),
+            );
+
+            expect(
+              await screen.findByRole("heading", {
+                name: dashboardInPersonalSubcollection.name,
+              }),
+            ).toBeInTheDocument();
+          });
+        });
+      });
+
+      describe("question is in a personal collection", () => {
+        it("should not render public collections", async () => {
+          await setup({
+            card: CARD_IN_PERSONAL_COLLECTION,
+            collections: COLLECTIONS,
+          });
+
+          expect(
+            screen.getByRole("heading", {
+              name: PERSONAL_COLLECTION.name,
+            }),
+          ).toBeInTheDocument();
+          expect(
+            screen.queryByRole("heading", {
+              name: COLLECTION.name,
+            }),
+          ).not.toBeInTheDocument();
+        });
+
+        describe('"Create a new dashboard" option', () => {
+          it('should not render "Create a new dashboard" option when opening the root collection (public collection)', async () => {
+            await setup({
+              card: CARD_IN_PERSONAL_COLLECTION,
+            });
+
+            expect(
+              screen.queryByRole("heading", {
+                name: "Create a new dashboard",
+              }),
+            ).not.toBeInTheDocument();
+          });
+
+          it('should render "Create a new dashboard" option when opening personal subcollections', async () => {
```

Comment:
test Query builder > Newly created question > #5 from [the epic](https://github.com/metabase/metabase/issues/34733)